vllm/v1/metrics/stats.py

```diff
@@ -151,3 +168,11 @@ def update_from_finished_request(self, finish_reason: "FinishReason",
                                  inference_time=inference_time,
                                  decode_time=decode_time)
         self.finished_requests.append(finished_req)
+
+        if lora_stats is not None:
```

Comment:
One state transition which may not be accounted for here is if a request is aborted

Aborts occur when the client disconnects from the server. The `AsyncLLM` sends an abort message to the `OutputProcessor` and to the `EngineCore`. This **may** cause the request to be finished, but I am not 100% sure that it does. This could cause the request to never be removed from `running_requests`. We should just double check this.

Additionally, if a client aborts when the request is in `lora_stats.waiting_requests` it will not ever be removed from the waiting set since the request will never be scheduled.

Reply:
Great point, take a look now - basically I've tried to couple this closely with `RequestState` lifecycle